### PR TITLE
Closes #348 | Implemented dataloader for indoner_tourism

### LINKED ISSUE #348: Create dataset loader for IndoNER-Tourism
Dataloader name: `indoner_tourism/indoner_tourism.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?indoner_tourism

|     Dataset| indoner_tourism |
|-------------|---|
| Description | This dataset is designed for named entity recognition (NER) tasks in the Bahasa Indonesia tourism domain. It contains labeled sequences of named entities, including locations, facilities, and tourism-related entities. The dataset is annotated with the following entity types:  - B-WIS: Beginning of a tourism-related entity. - I-WIS: Continuation of a tourism-related entity. - B-LOC: Beginning of a location entity. - I-LOC: Continuation of a location entity. - B-FAS: Beginning of a facility entity. - I-FAS: Continuation of a facility entity. - O: Non-entity or other words not falling into the specified categories. |
| Subsets | - |
| Languages | ind |
| Tasks | Named Entiy Recognition |
| License | Academic Free License v3.0 (afl-3.0) |
| Homepage | https://github.com/fathanick/IndoNER-Tourism/tree/main |
| HF URL | - |
| Paper URL | https://www.inacl.id/journal/index.php/jlk/article/download/89/63 |

### PR DESCRIPTION
**Pull request by luckysusanto**
Closes #348 

### Checkbox
- [X] Confirm that this PR is linked to the dataset issue.
- [X] Create the dataloader script `seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py` (please use only lowercase and underscore for dataset naming).
- [X] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [X] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [X] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [X] Confirm dataloader script works with `datasets.load_dataset` function.
- [X] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+}
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class IndoNERTourism(datasets.GeneratorBasedBuilder):
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py:86]:
```suggestion
class IndoNERTourismDataset(datasets.GeneratorBasedBuilder):
```

File: seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    B-FAS (5): Beginning of a facility entity.
+    I-FAS (6): Continuation of a facility entity.
+"""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name="indoner_tourism_source",
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py:104]: Let's use `_DATASETNAME` here (and also in the `subset_id`)

File: seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+_LANGUAGES = ['ind']  # We follow ISO639-3 language code (https://iso639-3.sil.org/code_tables/639/data)
+
+_LICENSE = Licenses.AFL_3_0.value 
+
+_LOCAL = False
+
+_URLS = {
+    _DATASETNAME: "https://raw.githubusercontent.com/fathanick/IndoNER-Tourism/main/ner_data.tsv",
+}
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py:77]: Nit: it's fine to use `_URL` here

```
_URL = "https://raw.githubusercontent.com/fathanick/IndoNER-Tourism/main/ner_data.tsv"
```

File: seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+            for line in file:
+                if line.strip() == "":
+                    if self.config.schema == "source":
+                        yield(
+                            counter,
+                            {
+                                'tokens'  : tokens,
+                                'ner_tags': ner_tags,
+                            }
+                        )
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py:172]: The use of `yield` here seems non-idiomatic. Perhaps we can do something like this:

```python
yield counter, {"tokens": tokens, "ner_tags": ner_tags}
```

Same comment in lines 177--184

File: seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+                        tokens = []
+                        ner_tags = []
+                elif len(line.split('\t')) == 2:
+                    token, ner_tag = line.split('\t')
+                    tokens.append(token.strip())
+                    if ner_tag not in ["O", "B-WIS", "I-WIS", "B-LOC", "I-LOC", "B-FAS", "I-FAS"]:
+                        if ner_tag[0] in ["B", "I"]:
+                            if any(tag in ner_tag for tag in ["WIS", "LOC", "FAS"]):
+                                if '_' in ner_tag:
+                                    ner_tag = '-'.join(ner_tag.split('_'))
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py:195]: Can you add a brief comment on what these lines do?

**Review by ljvmiranda921 — COMMENTED** [review]
Hi @luckysusanto , thanks for this PR! The dataloader works properly, but I have a few nits (as seen in the comments below).

Also, please rerun `make check_file=...` to ensure that the formatting is consistent with the other implementations. For example, there are unused imports in this data loader (`os`, `pandas`)

File: seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py (hunk 1 above)
Comment (luckysusanto) [reply seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py:86]: Implemented!

File: seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py (hunk 2 above)
Comment (luckysusanto) [reply seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py:104]: Implemented!

File: seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py (hunk 3 above)
Comment (luckysusanto) [reply seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py:77]: Implemented!

File: seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py (hunk 4 above)
Comment (luckysusanto) [reply seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py:172]: Resolved!

File: seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py (hunk 5 above)
Comment (luckysusanto) [reply seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py:195]: Comment added!

File: seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py (hunk 1 above)
Comment (luckysusanto) [reply seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py:86]: I have an issue with performing the make command :(, I'll try to find a solution to this, but, all I have attempted failed (even downloading wsl).

**Comment by luckysusanto** [comment]
Requesting review!
@holylovenia @ljvmiranda921

**Review by ljvmiranda921 — APPROVED** [review]
LGTM!

**Review by holylovenia — APPROVED** [review]
LGTM, @luckysusanto!! Thanks for your contribution. 😁

**Pull request merged**